sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -258,24 +258,30 @@ private RelNode possiblyWrapRootWithOuterLimitFromContext(
     if (root.rel instanceof Sort) {
       Sort innerSort = (Sort) root.rel;
       final int offset = Calcites.getOffset(innerSort);
+      final int innerLimit = Calcites.getFetch(innerSort);
       final int fetch = Calcites.collapseFetch(
-          Calcites.getFetch(innerSort),
+          innerLimit,
           Ints.checkedCast(outerLimit),
           0
       );
 
+      if (fetch == innerLimit) {
+        // nothing to do, don't bother to make a new sort
+        return root.rel;
+      }
+
+      root.rel.getCluster().getPlanner().setImportance(innerSort, 0.0);
```

Comment:
Can you add comment on what the `setImportance` call does?

Reply:
I don't actually think it is .. important to set here, I copied from sort collapse rule just to be safe, but the unit test and my manual test passes without it, so I don't believe it is necessary and have removed this line instead.

Reply:
Setting the importance of a rel to 0 is a way to tell the planner that it should not waste its time considering that rel: i.e., we are confident the new rel is better.